src/Nix/Exec.hs

```diff
@@ -570,10 +573,10 @@ instance (MonadFix m, MonadCatch m, MonadThrow m, MonadIO m,
         const $ toNix (0 :: Integer)
 #endif
 
-runLazyM :: Options -> MonadIO m => Lazy m a -> m a
-runLazyM opts = (`evalStateT` M.empty)
-              . (`runReaderT` newContext opts)
-              . runLazy
+runLazyM :: Options -> POSIXTime -> MonadIO m => Lazy m a -> m a
```

Comment:
This change is probably going to cause errors in the test suite.  I don't see any other places in this PR where you make changes to code that calls runLazyM.

Reply:
I’d recommend making the current time an option, which gives you the ability to set it on the command line for testing purposes. Then you won’t need to add a new entry to the Context (even though I know I suggested it before).

Also, I’m more familiar with seeing UTCTime used in Haskell code than POSIXTime.

Reply:
I probably led him astray there.  I too am more used to working with UTCTime, but the nix builtin wants POSIX time, so using Haskell's `POSIXTime` type seemed the most natural to me.  Your call.

Reply:
I'll make changes this evening. Thanks!